### ISSUE TITLE
Support vlan-filtering configuration

### ISSUE THREAD

**Issue by qinqon** [nmstate/kubernetes-nmstate#171]
Starting with version 0.3, nmstate allows users to configure [vlan-filtering options](https://github.com/nmstate/nmstate/pull/892), e.g.:

```yaml
interfaces:
  - name: linux-br0
    type: linux-bridge
    state: up
    bridge:
      port:
        - name: eth1
          vlan:
            mode: trunk
            trunk-tags:
            - id: 101
            - id-range:
                min: 500
                max: 599
            tag: 100
            enable-native: true
```

kubernetes-nmstate however handles the vlan-filtering configuration with a [workaround script](https://github.com/nmstate/kubernetes-nmstate/blob/master/build/bin/vlan-filtering) which makes sure to enable vlan-filtering by default and treat all bridge ports as trunks.

In order to introduce support for this feature while maintaining backwards compatibility, we should:

* Drop the custom script.
* In case no VLAN configuration is passed by the user, we should fill in defaults matching the behaviour of the script.

### Original description

Until we don't use NetworkManager to set vlan filtering (with or without nmstate) the linux bridge vlan_filtering flag is only set after desiredState is processed, so after reboot the flag is not going to be there until kubernetes-nmstate is not up and NodeNetworkState processed.

**Comment by jasinner** [comment]
Would this also allow us to remove hostNetworking SCC in OpenShift?

**Comment by qinqon** [comment]
@jasinner we think so, but it will take time, also I think the latest nmstate do some stuff out of NetworkManager dbus @phoracek do you remember what was it ?

**Comment by phoracek** [comment]
Underlying `nmstate` component we use to configure host networking indeed uses D-Bus to talk to NetworkManager, but it is not meant to be limited to it. `hostNetworking` will be always needed unless nmstate switches to "pure D-Bus" which I doubt. There will be always some additional work needed to do that NetworkManager may not support.

tl;dr: Even after we start using nmstate to set vlan_filtering, we will need to keep hostNetworking.

**Comment by qinqon** [comment]
We plan on implementing nmstate vlan-filtering level.
